### PR TITLE
SONARIAC-1508 Decouple 'Analyzer' from 'IacSensor'

### PR DESCRIPTION
**Pull request by mstachniuk**

### CODE REVIEW

File: iac-common/src/main/java/org/sonar/iac/common/extension/Analyzer.java [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+  private void analyseFile(InputFileContext inputFileContext) {
+    var inputFile = inputFileContext.inputFile;
+    String content;
+    try {
+      content = inputFile.contents();
+    } catch (IOException | RuntimeException e) {
+      throw ParseException.toParseException("read", inputFileContext, e);
+    }
+
+    if (content.isBlank()) {
```
Comment (jonas-wielage-sonarsource) [inline iac-common/src/main/java/org/sonar/iac/common/extension/Analyzer.java:83]: Nice improvement

Comment (jonas-wielage-sonarsource) [inline iac-common/src/main/java/org/sonar/iac/common/extension/Analyzer.java:83]: Nice improvement

File: iac-common/src/test/java/org/sonar/iac/common/extension/AnalyzerTest.java [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+    when(file.toString()).thenReturn("FileWithContent");
+    return file;
+  }
+
+  InputFile emptyFile() throws IOException {
+    InputFile file = mock(InputFile.class);
+    when(file.contents()).thenReturn("  ");
+    when(file.toString()).thenReturn("EmptyFile");
+    return file;
+  }
```
Comment (jonas-wielage-sonarsource) [inline iac-common/src/test/java/org/sonar/iac/common/extension/AnalyzerTest.java:190]: Why are you Mocking everything? Is it not possible to use the SensorContextTester or actual InputFile via `IacTestUtils`?

**Review by jonas-wielage-sonarsource — APPROVED** [review]
LGTM, please have a look at my comments

File: iac-common/src/test/java/org/sonar/iac/common/extension/AnalyzerTest.java (hunk 2 above)
Comment (mstachniuk) [reply iac-common/src/test/java/org/sonar/iac/common/extension/AnalyzerTest.java:190]: We will try tomorrow to improve it

**Pull request merged**